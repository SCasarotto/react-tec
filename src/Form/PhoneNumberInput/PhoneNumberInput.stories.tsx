import React from 'react';

import { Story, Meta } from '@storybook/react';

import { PhoneNumberInput, PhoneNumberInputProps } from './PhoneNumberInput';

export default {
  title: 'Form/PhoneNumberInput',
  component: PhoneNumberInput,
  argTypes: {},
} as Meta;

const Template: Story<PhoneNumberInputProps> = (args) => (
  <PhoneNumberInput {...args} />
);

export const Default = Template.bind({});
Default.args = {
  onChange: (phoneNumber) => console.log(phoneNumber),
};
