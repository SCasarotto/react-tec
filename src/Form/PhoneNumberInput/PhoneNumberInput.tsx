import React from 'react';

import PhoneInput, { PhoneInputProps } from 'react-phone-number-input';

import { StyledPhoneNumberInput } from './styledComponents';

export interface PhoneNumberInputProps extends PhoneInputProps {}
/**
 *
 * PhoneNumberInput is a simple styled [React Phone Number Input](https://gitlab.com/catamphetamine/react-phone-number-input) Component.
 *
 */
export const PhoneNumberInput = React.forwardRef<
  PhoneInput,
  PhoneNumberInputProps
>((props, ref) => {
  const { className = '', ...rest } = props;
  return (
    <StyledPhoneNumberInput
      className={`tec-phoneNumberInput ${className}`}
      ref={ref}
      {...rest}
    />
  );
});
