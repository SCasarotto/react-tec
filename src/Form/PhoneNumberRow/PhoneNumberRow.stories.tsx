import React from 'react';

import { Story, Meta } from '@storybook/react';

import { PhoneNumberRow, PhoneNumberRowProps } from './PhoneNumberRow';

export default {
  title: 'Form/PhoneNumberRow',
  component: PhoneNumberRow,
  argTypes: {},
} as Meta;

const Template: Story<PhoneNumberRowProps> = (args) => (
  <PhoneNumberRow {...args} />
);

export const Default = Template.bind({});
Default.args = {
  title: 'My Phone Number Input',
  labelForKey: 'SomeUniqueKey42',
  onChange: (phoneNumber) => console.log(phoneNumber),
};
