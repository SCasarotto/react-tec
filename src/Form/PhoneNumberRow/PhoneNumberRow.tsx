import React from 'react';

import PhoneInput from 'react-phone-number-input';

import { FormRow, FormRowCustomProps } from '../FormRow';
import { Label, LabelCustomProps } from '../Label';
import { PhoneNumberInput, PhoneNumberInputProps } from '../PhoneNumberInput';

export interface PhoneNumberRowProps
  extends FormRowCustomProps,
    LabelCustomProps,
    PhoneNumberInputProps {
  labelForKey: string;
}
/**
 *
 * A component containing a `Row`, a `Label` and either a `PhoneNumberInput`.
 *
 * In addition to the props below, reference all props for `PhoneNumberInput`.
 *
 */
export const PhoneNumberRow = React.forwardRef<PhoneInput, PhoneNumberRowProps>(
  (props, ref) => {
    const {
      rowSize,
      last,
      className = '',
      title,
      type,
      disabled,
      required,
      labelForKey,
      ...rest
    } = props;

    return (
      <FormRow
        rowSize={rowSize}
        last={last}
        className={`tec-phoneNumberRow ${className}`}
      >
        {title && (
          <Label
            htmlFor={labelForKey}
            required={required}
            disabled={disabled}
            className='tec-phoneNumberRowTitle'
          >
            {title}
          </Label>
        )}
        <PhoneNumberInput
          type={type}
          id={labelForKey}
          disabled={disabled}
          className='tec-phoneNumberRowInput'
          ref={ref}
          {...rest}
        />
      </FormRow>
    );
  },
);
