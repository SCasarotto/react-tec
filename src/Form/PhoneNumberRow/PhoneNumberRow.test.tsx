import React from 'react';

import { render, waitFor } from '@testing-library/react';
import PhoneNumberInput from 'react-phone-number-input';

import { PhoneNumberRow } from './PhoneNumberRow';

describe('PhoneNumberRow', () => {
  // Renders Without Error
  it('matches snapshot', () => {
    const { asFragment } = render(
      <PhoneNumberRow
        labelForKey='testPhoneNumberRow'
        onChange={jest.fn()}
        value=''
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  // Class Name
  it('accepts custom className', () => {
    const { container } = render(
      <PhoneNumberRow
        labelForKey='testPhoneNumberRow'
        onChange={jest.fn()}
        value=''
        className='myClassName'
      />,
    );
    expect(container.firstChild).toHaveClass('myClassName');
  });
  // Ref
  it('renders with ref', async () => {
    const ref: { current: PhoneNumberInput | null } = { current: null };
    render(
      <PhoneNumberRow
        labelForKey='testPhoneNumberRow'
        onChange={jest.fn()}
        value=''
        ref={ref}
      />,
    );
    waitFor(() => expect(ref.current).toBeTruthy());
  });
});
