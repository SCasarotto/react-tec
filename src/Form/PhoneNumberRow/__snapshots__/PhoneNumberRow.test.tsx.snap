// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneNumberRow matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  display: inline-block;
  vertical-align: top;
  margin-bottom: 10px;
  text-align: left;
  box-sizing: border-box;
  padding-right: 0px;
  width: 100%;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  font-size: 14px;
  box-sizing: border-box;
  cursor: text;
  border-radius: 5px;
  border: 1px solid;
  -webkit-transition: border-color 0.2s ease-in,box-shadow 0.2s ease-in;
  transition: border-color 0.2s ease-in,box-shadow 0.2s ease-in;
}

.c1:hover,
.c1:active {
  border: 1px solid;
  box-shadow: 0 0 0 1px inset;
}

.c1 .PhoneInputInput {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  min-width: 0;
  outline: none;
  border-radius: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
  font-size: inherit;
  margin: 0px;
  padding: 10px;
  background-color: transparent;
}

.c1 .PhoneInputCountryIcon {
  width: calc( var(--PhoneInputCountryFlag-height) * var(--PhoneInputCountryFlag-aspectRatio) );
  height: var(--PhoneInputCountryFlag-height);
  display: block;
}

.c1 .PhoneInputCountryIcon--square {
  width: var(--PhoneInputCountryFlag-height);
}

.c1 .PhoneInputCountryIcon--border {
  background-color: var(--PhoneInputCountryFlag-backgroundColor--loading);
  box-shadow: 0 0 0 var(--PhoneInputCountryFlag-borderWidth) var(--PhoneInputCountryFlag-borderColor), inset 0 0 0 var(--PhoneInputCountryFlag-borderWidth) var(--PhoneInputCountryFlag-borderColor);
}

.c1 .PhoneInputCountryIconImg {
  width: 100%;
  height: 100%;
}

.c1 .PhoneInputInternationalIconPhone {
  opacity: var(--PhoneInputInternationalIconPhone-opacity);
}

.c1 .PhoneInputInternationalIconGlobe {
  opacity: var(--PhoneInputInternationalIconGlobe-opacity);
}

.c1 .PhoneInputCountry {
  position: relative;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: var(--PhoneInputCountrySelect-marginRight);
  padding: 10px 0px 10px 10px;
}

.c1 .PhoneInputCountrySelect {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  z-index: 1;
  border: 0;
  opacity: 0;
  cursor: pointer;
}

.c1 .PhoneInputCountrySelect[disabled] {
  cursor: not-allowed;
}

.c1 .PhoneInputCountrySelectArrow {
  display: block;
  content: '';
  width: var(--PhoneInputCountrySelectArrow-width);
  height: var(--PhoneInputCountrySelectArrow-width);
  margin-top: var(--PhoneInputCountrySelectArrow-marginTop);
  margin-left: var(--PhoneInputCountrySelectArrow-marginLeft);
  border-style: solid;
  border-color: var(--PhoneInputCountrySelectArrow-color);
  border-top-width: 0;
  border-bottom-width: var(--PhoneInputCountrySelectArrow-borderWidth);
  border-left-width: 0;
  border-right-width: var(--PhoneInputCountrySelectArrow-borderWidth);
  -webkit-transform: var(--PhoneInputCountrySelectArrow-transform);
  -ms-transform: var(--PhoneInputCountrySelectArrow-transform);
  transform: var(--PhoneInputCountrySelectArrow-transform);
  opacity: var(--PhoneInputCountrySelectArrow-opacity);
}

.c1 .PhoneInputCountrySelect:focus + .PhoneInputCountryIcon + .PhoneInputCountrySelectArrow {
  opacity: 1;
  color: var(--PhoneInputCountrySelectArrow-color--focus);
}

.c1 .PhoneInputCountrySelect:focus + .PhoneInputCountryIcon--border {
  box-shadow: 0 0 0 var(--PhoneInputCountryFlag-borderWidth) var(--PhoneInputCountryFlag-borderColor--focus), inset 0 0 0 var(--PhoneInputCountryFlag-borderWidth) var(--PhoneInputCountryFlag-borderColor--focus);
}

.c1 .PhoneInputCountrySelect:focus + .PhoneInputCountryIcon .PhoneInputInternationalIconGlobe {
  opacity: 1;
  color: var(--PhoneInputCountrySelectArrow-color--focus);
}

<div
    class="c0 tec-formRow tec-phoneNumberRow "
  >
    <div
      class="c1 tec-phoneNumberInput tec-phoneNumberRowInput PhoneInput"
    >
      <div
        class="PhoneInputCountry"
      >
        <select
          aria-label="Phone number country"
          class="PhoneInputCountrySelect"
        >
          <option
            value="ZZ"
          >
            International
          </option>
          <option
            value="AF"
          >
            Afghanistan
          </option>
          <option
            value="AX"
          >
            Ã…land Islands
          </option>
          <option
            value="AL"
          >
            Albania
          </option>
          <option
            value="DZ"
          >
            Algeria
          </option>
          <option
            value="AS"
          >
            American Samoa
          </option>
          <option
            value="AD"
          >
            Andorra
          </option>
          <option
            value="AO"
          >
            Angola
          </option>
          <option
            value="AI"
          >
            Anguilla
          </option>
          <option
            value="AG"
          >
            Antigua and Barbuda
          </option>
          <option
            value="AR"
          >
            Argentina
          </option>
          <option
            value="AM"
          >
            Armenia
          </option>
          <option
            value="AW"
          >
            Aruba
          </option>
          <option
            value="AC"
          >
            Ascension Island
          </option>
          <option
            value="AU"
          >
            Australia
          </option>
          <option
            value="AT"
          >
            Austria
          </option>
          <option
            value="AZ"
          >
            Azerbaijan
          </option>
          <option
            value="BS"
          >
            Bahamas
          </option>
          <option
            value="BH"
          >
            Bahrain
          </option>
          <option
            value="BD"
          >
            Bangladesh
          </option>
          <option
            value="BB"
          >
            Barbados
          </option>
          <option
            value="BY"
          >
            Belarus
          </option>
          <option
            value="BE"
          >
            Belgium
          </option>
          <option
            value="BZ"
          >
            Belize
          </option>
          <option
            value="BJ"
          >
            Benin
          </option>
          <option
            value="BM"
          >
            Bermuda
          </option>
          <option
            value="BT"
          >
            Bhutan
          </option>
          <option
            value="BO"
          >
            Bolivia
          </option>
          <option
            value="BQ"
          >
            Bonaire, Sint Eustatius and Saba
          </option>
          <option
            value="BA"
          >
            Bosnia and Herzegovina
          </option>
          <option
            value="BW"
          >
            Botswana
          </option>
          <option
            value="BR"
          >
            Brazil
          </option>
          <option
            value="IO"
          >
            British Indian Ocean Territory
          </option>
          <option
            value="BN"
          >
            Brunei Darussalam
          </option>
          <option
            value="BG"
          >
            Bulgaria
          </option>
          <option
            value="BF"
          >
            Burkina Faso
          </option>
          <option
            value="MM"
          >
            Burma
          </option>
          <option
            value="BI"
          >
            Burundi
          </option>
          <option
            value="KH"
          >
            Cambodia
          </option>
          <option
            value="CM"
          >
            Cameroon
          </option>
          <option
            value="CA"
          >
            Canada
          </option>
          <option
            value="CV"
          >
            Cape Verde
          </option>
          <option
            value="KY"
          >
            Cayman Islands
          </option>
          <option
            value="CF"
          >
            Central African Republic
          </option>
          <option
            value="TD"
          >
            Chad
          </option>
          <option
            value="CL"
          >
            Chile
          </option>
          <option
            value="CN"
          >
            China
          </option>
          <option
            value="CX"
          >
            Christmas Island
          </option>
          <option
            value="CC"
          >
            Cocos (Keeling) Islands
          </option>
          <option
            value="CO"
          >
            Colombia
          </option>
          <option
            value="KM"
          >
            Comoros
          </option>
          <option
            value="CG"
          >
            Congo
          </option>
          <option
            value="CD"
          >
            Congo, Democratic Republic of the
          </option>
          <option
            value="CK"
          >
            Cook Islands
          </option>
          <option
            value="CR"
          >
            Costa Rica
          </option>
          <option
            value="CI"
          >
            Cote d'Ivoire
          </option>
          <option
            value="HR"
          >
            Croatia
          </option>
          <option
            value="CU"
          >
            Cuba
          </option>
          <option
            value="CW"
          >
            CuraÃ§ao
          </option>
          <option
            value="CY"
          >
            Cyprus
          </option>
          <option
            value="CZ"
          >
            Czech Republic
          </option>
          <option
            value="DK"
          >
            Denmark
          </option>
          <option
            value="DJ"
          >
            Djibouti
          </option>
          <option
            value="DM"
          >
            Dominica
          </option>
          <option
            value="DO"
          >
            Dominican Republic
          </option>
          <option
            value="EC"
          >
            Ecuador
          </option>
          <option
            value="EG"
          >
            Egypt
          </option>
          <option
            value="SV"
          >
            El Salvador
          </option>
          <option
            value="GQ"
          >
            Equatorial Guinea
          </option>
          <option
            value="ER"
          >
            Eritrea
          </option>
          <option
            value="EE"
          >
            Estonia
          </option>
          <option
            value="ET"
          >
            Ethiopia
          </option>
          <option
            value="FK"
          >
            Falkland Islands
          </option>
          <option
            value="FO"
          >
            Faroe Islands
          </option>
          <option
            value="FM"
          >
            Federated States of Micronesia
          </option>
          <option
            value="FJ"
          >
            Fiji
          </option>
          <option
            value="FI"
          >
            Finland
          </option>
          <option
            value="FR"
          >
            France
          </option>
          <option
            value="GF"
          >
            French Guiana
          </option>
          <option
            value="PF"
          >
            French Polynesia
          </option>
          <option
            value="GA"
          >
            Gabon
          </option>
          <option
            value="GM"
          >
            Gambia
          </option>
          <option
            value="GE"
          >
            Georgia
          </option>
          <option
            value="DE"
          >
            Germany
          </option>
          <option
            value="GH"
          >
            Ghana
          </option>
          <option
            value="GI"
          >
            Gibraltar
          </option>
          <option
            value="GR"
          >
            Greece
          </option>
          <option
            value="GL"
          >
            Greenland
          </option>
          <option
            value="GD"
          >
            Grenada
          </option>
          <option
            value="GP"
          >
            Guadeloupe
          </option>
          <option
            value="GU"
          >
            Guam
          </option>
          <option
            value="GT"
          >
            Guatemala
          </option>
          <option
            value="GG"
          >
            Guernsey
          </option>
          <option
            value="GN"
          >
            Guinea
          </option>
          <option
            value="GW"
          >
            Guinea-Bissau
          </option>
          <option
            value="GY"
          >
            Guyana
          </option>
          <option
            value="HT"
          >
            Haiti
          </option>
          <option
            value="VA"
          >
            Holy See (Vatican City State)
          </option>
          <option
            value="HN"
          >
            Honduras
          </option>
          <option
            value="HK"
          >
            Hong Kong
          </option>
          <option
            value="HU"
          >
            Hungary
          </option>
          <option
            value="IS"
          >
            Iceland
          </option>
          <option
            value="IN"
          >
            India
          </option>
          <option
            value="ID"
          >
            Indonesia
          </option>
          <option
            value="IR"
          >
            Iran
          </option>
          <option
            value="IQ"
          >
            Iraq
          </option>
          <option
            value="IE"
          >
            Ireland
          </option>
          <option
            value="IM"
          >
            Isle of Man
          </option>
          <option
            value="IL"
          >
            Israel
          </option>
          <option
            value="IT"
          >
            Italy
          </option>
          <option
            value="JM"
          >
            Jamaica
          </option>
          <option
            value="JP"
          >
            Japan
          </option>
          <option
            value="JE"
          >
            Jersey
          </option>
          <option
            value="JO"
          >
            Jordan
          </option>
          <option
            value="KZ"
          >
            Kazakhstan
          </option>
          <option
            value="KE"
          >
            Kenya
          </option>
          <option
            value="KI"
          >
            Kiribati
          </option>
          <option
            value="XK"
          >
            Kosovo
          </option>
          <option
            value="KW"
          >
            Kuwait
          </option>
          <option
            value="KG"
          >
            Kyrgyzstan
          </option>
          <option
            value="LA"
          >
            Laos
          </option>
          <option
            value="LV"
          >
            Latvia
          </option>
          <option
            value="LB"
          >
            Lebanon
          </option>
          <option
            value="LS"
          >
            Lesotho
          </option>
          <option
            value="LR"
          >
            Liberia
          </option>
          <option
            value="LY"
          >
            Libya
          </option>
          <option
            value="LI"
          >
            Liechtenstein
          </option>
          <option
            value="LT"
          >
            Lithuania
          </option>
          <option
            value="LU"
          >
            Luxembourg
          </option>
          <option
            value="MO"
          >
            Macao
          </option>
          <option
            value="MG"
          >
            Madagascar
          </option>
          <option
            value="MW"
          >
            Malawi
          </option>
          <option
            value="MY"
          >
            Malaysia
          </option>
          <option
            value="MV"
          >
            Maldives
          </option>
          <option
            value="ML"
          >
            Mali
          </option>
          <option
            value="MT"
          >
            Malta
          </option>
          <option
            value="MH"
          >
            Marshall Islands
          </option>
          <option
            value="MQ"
          >
            Martinique
          </option>
          <option
            value="MR"
          >
            Mauritania
          </option>
          <option
            value="MU"
          >
            Mauritius
          </option>
          <option
            value="YT"
          >
            Mayotte
          </option>
          <option
            value="MX"
          >
            Mexico
          </option>
          <option
            value="MD"
          >
            Moldova
          </option>
          <option
            value="MC"
          >
            Monaco
          </option>
          <option
            value="MN"
          >
            Mongolia
          </option>
          <option
            value="ME"
          >
            Montenegro
          </option>
          <option
            value="MS"
          >
            Montserrat
          </option>
          <option
            value="MA"
          >
            Morocco
          </option>
          <option
            value="MZ"
          >
            Mozambique
          </option>
          <option
            value="NA"
          >
            Namibia
          </option>
          <option
            value="NR"
          >
            Nauru
          </option>
          <option
            value="NP"
          >
            Nepal
          </option>
          <option
            value="NL"
          >
            Netherlands
          </option>
          <option
            value="NC"
          >
            New Caledonia
          </option>
          <option
            value="NZ"
          >
            New Zealand
          </option>
          <option
            value="NI"
          >
            Nicaragua
          </option>
          <option
            value="NE"
          >
            Niger
          </option>
          <option
            value="NG"
          >
            Nigeria
          </option>
          <option
            value="NU"
          >
            Niue
          </option>
          <option
            value="NF"
          >
            Norfolk Island
          </option>
          <option
            value="KP"
          >
            North Korea
          </option>
          <option
            value="MK"
          >
            North Macedonia
          </option>
          <option
            value="MP"
          >
            Northern Mariana Islands
          </option>
          <option
            value="NO"
          >
            Norway
          </option>
          <option
            value="OM"
          >
            Oman
          </option>
          <option
            value="PK"
          >
            Pakistan
          </option>
          <option
            value="PW"
          >
            Palau
          </option>
          <option
            value="PS"
          >
            Palestine
          </option>
          <option
            value="PA"
          >
            Panama
          </option>
          <option
            value="PG"
          >
            Papua New Guinea
          </option>
          <option
            value="PY"
          >
            Paraguay
          </option>
          <option
            value="PE"
          >
            Peru
          </option>
          <option
            value="PH"
          >
            Philippines
          </option>
          <option
            value="PL"
          >
            Poland
          </option>
          <option
            value="PT"
          >
            Portugal
          </option>
          <option
            value="PR"
          >
            Puerto Rico
          </option>
          <option
            value="QA"
          >
            Qatar
          </option>
          <option
            value="RE"
          >
            Reunion
          </option>
          <option
            value="RO"
          >
            Romania
          </option>
          <option
            value="RU"
          >
            Russia
          </option>
          <option
            value="RW"
          >
            Rwanda
          </option>
          <option
            value="BL"
          >
            Saint BarthÃ©lemy
          </option>
          <option
            value="SH"
          >
            Saint Helena
          </option>
          <option
            value="KN"
          >
            Saint Kitts and Nevis
          </option>
          <option
            value="LC"
          >
            Saint Lucia
          </option>
          <option
            value="MF"
          >
            Saint Martin (French Part)
          </option>
          <option
            value="PM"
          >
            Saint Pierre and Miquelon
          </option>
          <option
            value="VC"
          >
            Saint Vincent and the Grenadines
          </option>
          <option
            value="WS"
          >
            Samoa
          </option>
          <option
            value="SM"
          >
            San Marino
          </option>
          <option
            value="ST"
          >
            Sao Tome and Principe
          </option>
          <option
            value="SA"
          >
            Saudi Arabia
          </option>
          <option
            value="SN"
          >
            Senegal
          </option>
          <option
            value="RS"
          >
            Serbia
          </option>
          <option
            value="SC"
          >
            Seychelles
          </option>
          <option
            value="SL"
          >
            Sierra Leone
          </option>
          <option
            value="SG"
          >
            Singapore
          </option>
          <option
            value="SX"
          >
            Sint Maarten
          </option>
          <option
            value="SK"
          >
            Slovakia
          </option>
          <option
            value="SI"
          >
            Slovenia
          </option>
          <option
            value="SB"
          >
            Solomon Islands
          </option>
          <option
            value="SO"
          >
            Somalia
          </option>
          <option
            value="ZA"
          >
            South Africa
          </option>
          <option
            value="KR"
          >
            South Korea
          </option>
          <option
            value="SS"
          >
            South Sudan
          </option>
          <option
            value="ES"
          >
            Spain
          </option>
          <option
            value="LK"
          >
            Sri Lanka
          </option>
          <option
            value="SD"
          >
            Sudan
          </option>
          <option
            value="SR"
          >
            Suriname
          </option>
          <option
            value="SJ"
          >
            Svalbard and Jan Mayen
          </option>
          <option
            value="SZ"
          >
            Swaziland
          </option>
          <option
            value="SE"
          >
            Sweden
          </option>
          <option
            value="CH"
          >
            Switzerland
          </option>
          <option
            value="SY"
          >
            Syria
          </option>
          <option
            value="TW"
          >
            Taiwan
          </option>
          <option
            value="TJ"
          >
            Tajikistan
          </option>
          <option
            value="TZ"
          >
            Tanzania
          </option>
          <option
            value="TH"
          >
            Thailand
          </option>
          <option
            value="TL"
          >
            Timor-Leste
          </option>
          <option
            value="TG"
          >
            Togo
          </option>
          <option
            value="TK"
          >
            Tokelau
          </option>
          <option
            value="TO"
          >
            Tonga
          </option>
          <option
            value="TT"
          >
            Trinidad and Tobago
          </option>
          <option
            value="TA"
          >
            Tristan da Cunha
          </option>
          <option
            value="TN"
          >
            Tunisia
          </option>
          <option
            value="TR"
          >
            Turkey
          </option>
          <option
            value="TM"
          >
            Turkmenistan
          </option>
          <option
            value="TC"
          >
            Turks and Caicos Islands
          </option>
          <option
            value="TV"
          >
            Tuvalu
          </option>
          <option
            value="UG"
          >
            Uganda
          </option>
          <option
            value="UA"
          >
            Ukraine
          </option>
          <option
            value="AE"
          >
            United Arab Emirates
          </option>
          <option
            value="GB"
          >
            United Kingdom
          </option>
          <option
            value="US"
          >
            United States
          </option>
          <option
            value="UY"
          >
            Uruguay
          </option>
          <option
            value="UZ"
          >
            Uzbekistan
          </option>
          <option
            value="VU"
          >
            Vanuatu
          </option>
          <option
            value="VE"
          >
            Venezuela
          </option>
          <option
            value="VN"
          >
            Vietnam
          </option>
          <option
            value="VG"
          >
            Virgin Islands, British
          </option>
          <option
            value="VI"
          >
            Virgin Islands, U.S.
          </option>
          <option
            value="WF"
          >
            Wallis and Futuna
          </option>
          <option
            value="EH"
          >
            Western Sahara
          </option>
          <option
            value="YE"
          >
            Yemen
          </option>
          <option
            value="ZM"
          >
            Zambia
          </option>
          <option
            value="ZW"
          >
            Zimbabwe
          </option>
        </select>
        <div
          class="PhoneInputCountryIcon"
        >
          <svg
            class="PhoneInputCountryIconImg"
            viewBox="0 0 75 50"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              International
            </title>
            <g
              class="PhoneInputInternationalIconGlobe"
              fill="none"
              stroke="currentColor"
              stroke-miterlimit="10"
              stroke-width="2"
            >
              <path
                d="M47.2,36.1C48.1,36,49,36,50,36c7.4,0,14,1.7,18.5,4.3"
                stroke-linecap="round"
              />
              <path
                d="M68.6,9.6C64.2,12.3,57.5,14,50,14c-7.4,0-14-1.7-18.5-4.3"
              />
              <line
                x1="26"
                x2="74"
                y1="25"
                y2="25"
              />
              <line
                x1="50"
                x2="50"
                y1="1"
                y2="49"
              />
              <path
                d="M46.3,48.7c1.2,0.2,2.5,0.3,3.7,0.3c13.3,0,24-10.7,24-24S63.3,1,50,1S26,11.7,26,25c0,2,0.3,3.9,0.7,5.8"
                stroke-linecap="round"
              />
              <path
                d="M46.8,48.2c1,0.6,2.1,0.8,3.2,0.8c6.6,0,12-10.7,12-24S56.6,1,50,1S38,11.7,38,25c0,1.4,0.1,2.7,0.2,4c0,0.1,0,0.2,0,0.2"
                stroke-linecap="round"
              />
            </g>
            <path
              class="PhoneInputInternationalIconPhone"
              d="M12.4,17.9c2.9-2.9,5.4-4.8,0.3-11.2S4.1,5.2,1.3,8.1C-2,11.4,1.1,23.5,13.1,35.6s24.3,15.2,27.5,11.9c2.8-2.8,7.8-6.3,1.4-11.5s-8.3-2.6-11.2,0.3c-2,2-7.2-2.2-11.7-6.7S10.4,19.9,12.4,17.9z"
              fill="currentColor"
              stroke="none"
            />
          </svg>
        </div>
        <div
          class="PhoneInputCountrySelectArrow"
        />
      </div>
      <input
        autocomplete="tel"
        class="PhoneInputInput"
        id="testPhoneNumberRow"
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
